pkg/registry/apis/provisioning/repository/go-git/wrapper.go

```diff
@@ -83,7 +106,14 @@ func Clone(
 	if err != nil {
 		return nil, err
 	}
-	url := fmt.Sprintf("%s.git", gitcfg.URL)
+
+	// Create a size-limited writer that will cancel the context if size is exceeded
+	var progressWriter io.Writer
+	if opts.MaxSize > 0 {
+		progressWriter = newMaxBytesWriter(progress, opts.MaxSize, cancel)
```

Comment:
the writer here is reading each line of progress (stdout) from the job -- it is *not* the checkout itself, so the 1GB limit is not really relevant (nor would 1GB of output matter since it is all thrown away... yes other things have likely gone wrong if we got 1GB of stdout!)

Reply:
soo f*** blind! thank you!